eng/Directory.Build.Common.targets

```diff
@@ -330,5 +330,33 @@
     <Message Condition="'$(IsShippingLibrary)' == 'true'" Text="'$(PackageRootDirectory)' '$(ServiceDirectory)' '$(PackageId)' '$(VersionForProperties)' '$(PackageSdkType)' '$(PackageIsNewSdk)'" Importance="High" />
   </Target>
 
+  <Target Name="_SetProjectDependsOnInnerTarget">
+    <PropertyGroup>
+      <InnerTargets>ProjectDependsOnInner</InnerTargets>
+    </PropertyGroup>
+  </Target>
+
+  <Target
+    Name="ProjectDependsOn"
+    DependsOnTargets="_SetProjectDependsOnInnerTarget;DispatchToInnerBuilds"
+    Condition="'$(IsCrossTargetingBuild)' == 'true'"
+    Returns="@(ProjectsToInclude)">
+
+    <RemoveDuplicates Inputs="@(InnerOutput)">
+      <Output TaskParameter="Filtered" ItemName="ProjectsToInclude"/>
+    </RemoveDuplicates>
+  </Target>
+
+  <Target Name="ProjectDependsOnInner" DependsOnTargets="ResolveReferences" Returns="@(ProjectsToInclude)">
+    <ItemGroup>
+      <_LibrariesReferenced Include="@(ReferencePath -> '%(Filename)')"/>
+      <_DependsOnGiven Include="@(_LibrariesReferenced)" Condition="'%(Identity)' == '$(TestDependsOnDependency)'" />
+    </ItemGroup>
+
+    <ItemGroup Condition="'@(_DependsOnGiven)' != '' and '$(IsClientLibrary)' == 'true'" >
+      <ProjectsToInclude Include="$(MSBuildProjectFullPath)" />
+    </ItemGroup>
```

Comment:
Not a request for change in this PR, necessarily, but what if we do something like this to decide when to set `IsClientLibrary`? Currently, we do so if it's Azure.*, but we have some extensions that are Microsoft.Azure.* to avoid being confused with track 2 client SDKs, but are still treated like track 2 SDKs when it comes to building. They also depend on Azure.Core. IT should be true that everything that depends on Azure.Core is a track 2 SDK and, hence, `IsClientLibrary = true`. /cc @JoshLove-msft

Reply:
I'm not sure that every library depending on Core should be treated as a client library, but it might be worth exploring.

For the extensions, we have a different property, IsExtensionsLibrary.